designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/IconCardInput.kt

```diff
@@ -0,0 +1,344 @@
+package org.hisp.dhis.mobile.ui.designsystem.component
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import org.hisp.dhis.mobile.ui.designsystem.component.IconCardLayoutType.Matrix
+import org.hisp.dhis.mobile.ui.designsystem.component.IconCardLayoutType.Sequential
+import org.hisp.dhis.mobile.ui.designsystem.component.InputShellState.UNFOCUSED
+import org.hisp.dhis.mobile.ui.designsystem.component.internal.VerticalGrid
+import org.hisp.dhis.mobile.ui.designsystem.resource.provideDHIS2Icon
+import org.hisp.dhis.mobile.ui.designsystem.theme.DHISShapes
+import org.hisp.dhis.mobile.ui.designsystem.theme.Outline
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+import org.hisp.dhis.mobile.ui.designsystem.theme.bottomShadow
+
+/**
+ * DHIS2 icon card input component
+ * @param title: Label of the component
+ * @param data: List of [IconCardData] to show
+ * @param layoutType: [IconCardLayoutType] to render list in, Matrix or Sequential. Default is [IconCardLayoutType.Matrix]
+ * @param selectedData: Selected [IconCardData], renders selected UI around that item
+ * @param state: [InputShellState]
+ * @param supportingText: List of [SupportingTextData] that manages all the messages to be shown
+ * @param legendData: [LegendData] to be render below the input shell
+ * @param isRequired: Mark this input as marked
+ * @param onSelectionChanged: Callback to receive new selected item
+ */
+@Composable
+fun IconCardInput(
```

Comment:
As naming convention we are using `InputValueType` like InputText, InputNumber, so why not split it in InputMatrix and InputSequential?
![image](https://github.com/dhis2/dhis2-mobile-ui/assets/3330344/28ea9616-9751-477e-8662-eb3b26851752)

Reply:
Initially tried `InputIconCard` but grammatically it felt wrong for the component name. Since it's a icon card based input. But I can change it to `InputIconCard` if you think that makes more sense.

Reply:
I think that for a Dhis2 user would make more sense having it separate `InputMatrix` and `InputSequential`

Reply:
We are duplicating the `InputShell` usage for it in both the functions, that's why I extracted out a common function, but I guess that's fine?